### DISCUSSION TITLE
A Plugin To Colour Text

### DISCUSSION THREAD

**Discussion by KaptianCore** [remarkjs/remark#526] [Ideas] [answered]
I would like it if we had a plugin where I could do like 
[color="hex, rgb or worded colors"]text[/color]

would be very useful for colouring text

**Comment by ChristianMurphy** [comment, marked as answer]
This syntax isn't part of commonmark, but can be added through a plugin, like https://github.com/zestedesavoir/zmarkdown/tree/HEAD/packages/remark-custom-blocks#readme or https://github.com/djm/remark-shortcodes#readme.
once the plugin is parsing the new syntax, you could then add a transform to apply the color, see https://unifiedjs.com/learn for an intro to syntax trees and creating a plugin/transform.
